code/game/gamemodes/blob/theblob.dm

```diff
@@ -124,24 +128,33 @@
 			else	T = null
 
 	if(!T)	return 0
-	var/obj/effect/blob/B = new /obj/effect/blob/normal(src.loc)
+	//We can keep the above code in this proc, it probably works fine?
+	//new blob code with less BS:
+	var/Blob_spawnable = 1
 	if(istype(T, /turf/space) && prob(65))
-		B.health = 0
-	B.color = a_color
-	B.density = 1
-	if(T.Enter(B,src))//Attempt to move into the tile
-		B.density = initial(B.density)
-		B.loc = T
-		B.update_icon()
-	else
-		T.blob_act()//If we cant move in hit the turf
-		B.loc = null //So we don't play the splat sound, see Destroy()
-		qdel(B)
-
-	for(var/atom/A in T)//Hit everything in the turf
-		A.blob_act()
+		Blob_spawnable = 0
+		playsound(src.loc, 'sound/effects/splat.ogg', 50, 1) //Let's give some feedback that we DID try to spawn in space, since players are used to it
+	for (var/atom/A in T)
+		if (A.density != 0) // Unless density is 0, don't spawn a blob
+			Blob_spawnable = 0
+		A.blob_act() //Hit everything
+	if (T.density != 0) //Check for walls and such dense turfs
+		Blob_spawnable = 0
+		T.blob_act() //Hit the turf
+	if (Blob_spawnable == 1)
+		var/obj/effect/blob/B = new /obj/effect/blob/normal(src.loc)
+		B.color = a_color
+		B.density = 1
+		if(T.Enter(B,src)) //NOW we can attempt to move into the tile
+			B.density = initial(B.density)
+			B.loc = T
+			B.update_icon()
+		else
+			T.blob_act() //If we cant move in hit the turf
+			qdel(B) //We should never get to this point, since we checked before moving in. Destroy blob anyway for cleanliness though

```

Comment:
Wait, which comment? The one on qdel at the end?
I figured it shouldn't ever get to that else statement, since if there's nothing dense in the way, it should be able to move in?
If there IS another case (is there something else that stops it?), I would like to know, please, as I would like to put a check for it first ^^.